<!DOCTYPE html>
<!-- saved from url=(0080)https://unlimited.ethz.ch/plugins/viewsource/viewpagesrc.action?pageId=128713540 -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <title>View Source</title>
        <link rel="canonical" href="https://unlimited.ethz.ch/pages/viewpage.action?pageId=$action.page.id">
        <script>
window.WRM=window.WRM||{};window.WRM._unparsedData=window.WRM._unparsedData||{};window.WRM._unparsedErrors=window.WRM._unparsedErrors||{};
WRM._unparsedData["com.atlassian.plugins.atlassian-plugins-webresource-plugin:context-path.context-path"]="\u0022\u0022";
if(window.WRM._dataArrived)window.WRM._dataArrived();</script>
<link rel="stylesheet" href="./reference_based_phasing_files/batch.css" data-wrm-key="_super" data-wrm-batch-type="context" media="all">
<link rel="stylesheet" href="./reference_based_phasing_files/batch(1).css" data-wrm-key="plugin.viewsource,-_super" data-wrm-batch-type="context" media="all">
<link rel="stylesheet" href="./reference_based_phasing_files/batch(2).css" data-wrm-key="page,-_super" data-wrm-batch-type="context" media="all">
<link rel="stylesheet" href="./reference_based_phasing_files/batch(3).css" data-wrm-key="editor-content,-_super" data-wrm-batch-type="context" media="all">
<link rel="stylesheet" href="./reference_based_phasing_files/custom.css" media="all">

    </head>

    <body class="mceContentBody aui-theme-default wiki-content fullsize">
        <p>&nbsp;</p>         <p>I think now it's time to summarize the workflow for read-based phasing using WHATSHAP and HAPCUT2 with ONT long reads and Hi-C data.</p><p>Some introduction to read-based phasing or reference-based phasing can be seen here, <a href="https://www.sciencedirect.com/science/article/pii/S2001037019303836">https://www.sciencedirect.com/science/article/pii/S2001037019303836</a> and <a href="https://www.nature.com/articles/s41587-020-0711-0">https://www.nature.com/articles/s41587-020-0711-0</a>.&nbsp;</p><p><img class="confluence-embedded-image confluence-external-resource" draggable="false" alt="Fig. 1" src="./reference_based_phasing_files/41587_2020_711_Fig1_HTML.png" data-image-src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41587-020-0711-0/MediaObjects/41587_2020_711_Fig1_HTML.png"></p><p>As we can see from above graph copied from <a href="https://www.nature.com/articles/s41587-020-0711-0">https://www.nature.com/articles/s41587-020-0711-0</a>, the idea of read-based/reference-based phasing is following: first build unphased chromosomes, then use unphased chromosomes as reference, mapping short reads and long reads to the reference to derive chromosome-level phase blocks. Next, based on the chromosome-level phase block, long reads are split into two haplotype groups and finally each group of reads is assembled separately to reconstruct both homologous chromosomes. Any two SNPs in the chromosome covered by long reads will be phased, and many local phase blocks will be formed if there are no SNPs shared between long reads to extend the phase. Additional long-range linkage data, such as Hi-C reads, phased genetic maps, and single-strand seq, can provide global phase information to connect local phase blocks to form a chromosome-level phase block. However, it is normal that some small phase blocks cannot be integrated into the large phase block (<a href="https://www.nature.com/articles/s41587-020-0719-5">https://www.nature.com/articles/s41587-020-0719-5</a>) if the global phase data doesn't share any SNPs with the small block. For example, if there is no Hi-C read mapped to a small phase block,&nbsp; then this block cannot be integrated into the large phase block.</p><p><br></p><p>For our case, we developed our own workflow for read-based phasing with ONT data, but the workflow by definition should be able to work with other long reads, too:</p><p><img class="confluence-embedded-image" draggable="false" src="./reference_based_phasing_files/read_based_phasing_workflow.png" data-image-src="/download/attachments/128713540/read_based_phasing_workflow.png?version=1&amp;modificationDate=1661254013951&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128713614" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="read_based_phasing_workflow.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; read_based_phasing_workflow.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; read_based_phasing_workflow.png" data-image-height="557" data-image-width="436"></p><p>The idea of our workflow is, first use hapcut2 to build global phase blocks with Hi-C data based on the unphased chromosome-scale assembly, then incorporate the global phase information into whatshap with long reads to produce more dense large phase blocks. Ideally, there should be only one chromosome-level phase block for one chromosome. Whatshap will bin reads to different haplotype groups based on the chromosome-level phase block and then an assembler of choice, for example, flye can be used to quickly assemble each group of reads. After assembly, chimeric contigs are manually corrected by tritex pipeline with the help of Hi-C data,&nbsp; and finally pseudo-chromosomes are constructed by aligning corrected contigs to the unphased reference using a reference-guided scaffolding method, ragtag.</p><p><br></p><p>Now, step by step to do read-based phasing:</p><p>First step, building phase blocks with Hi-C using hapcut2</p><p>I made a new conda environment for hapcut2:</p><p><strong>conda create -n hapcut2</strong></p><p>Install hapcut2 by conda:</p><p><span style="color: rgb(51,51,51);"><strong>conda install -c bioconda hapcut2</strong> # actually, this installs dependencies required by hapcut2. Hapcut2 itself is a pipeline with the phase algorithms (which are developed by the hapcut2 team)</span></p><p>Also clone the github repository:</p><p><strong>git clone <a href="https://github.com/vibansal/HapCUT2.git">https://github.com/vibansal/HapCUT2.git</a></strong></p><p>Hapcut2 team provides a snakemake pipeline to run all the analyses required, so we just need to modify the input accordingly in the snakemake file, here's my snakemake file:</p><p><a class="confluence-link" href="https://unlimited.ethz.ch/download/attachments/128713540/Snakefile?version=1&amp;modificationDate=1661344408158&amp;api=v2" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" data-linked-resource-content-type="application/octet-stream" data-linked-resource-id="128714527" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="Snakefile" data-base-url="https://unlimited.ethz.ch">Snakefile</a>&nbsp;</p><p>To run snakemake pipeline, do this:</p><p><strong>snakemake -s Snakefile --verbose -p</strong>&nbsp;</p><p>There are many output files, but what we need is the phased vcf file for every chromosome. Then we remove unphased snps from each vcf and finally concatenate vcf files together</p><p><strong>#!/bin/bash</strong></p><p><strong># extract phased SNPs and combine all the phased SNPs</strong></p><p><strong>ls *.VCF &gt; VCF.list</strong></p><p><strong># extract phased SNPs</strong><br><strong>cat VCF.list | parallel -j 7 -k "bcftools view -p {} &gt; {}.phased.vcf"</strong></p><p><strong># concatenate phased SNPs</strong><br><strong>bcftools concat *.phased.vcf &gt; Rabiosa_hapcut2_phased_chr.vcf</strong></p><p><strong># compress vcf</strong><br><strong>bgzip -c Rabiosa_hapcut2_phased_chr.vcf &gt; Rabiosa_hapcut2_phased_chr.vcf.gz</strong><br><strong>bcftools index Rabiosa_hapcut2_phased_chr.vcf.gz</strong></p><p><strong># reheader vcf, change sample name</strong><br><strong>bcftools reheader -s <a href="http://sample.name/">sample.name</a> Rabiosa_hapcut2_phased_chr.vcf.gz &gt; Rabiosa_hapcut2_phased_chr_renamed.vcf.gz</strong></p><p>The content of sample.name is Rabiosa. The aim of renaming sample name in the vcf file is to make all vcf files incorporated in whatshap have the same sample name so whatshap can recognize the same sample in different file. This is important! If whatshap doesn't work for you, check the sample name in vcf file!</p><p><br></p><p>Before start whatshap, another thing we need to do is find SNPs in the reference. As it is more reliable to call SNPs with accurate short pair-end reads than error-prone ONT long reads, we use a BWA mem + bcftools approach to call SNPs with pair-end reads:</p><p><strong>#!/bin/bash</strong></p><p><strong># align PE700 reads to chromosome-level assembly&nbsp;</strong></p><p><strong>myread="/home/yutachen/public/Yutangchen/Rabiosa_data/tmp"</strong><br><strong>fasta="211209_Rabiosa_new_pseudomolecules_v1+unanchored_contigs.fasta"</strong></p><p><strong># index the assembly</strong><br><strong>bwa index -a bwtsw reference/${fasta}</strong></p><p><strong># align PE700 reads of rabiosa to the assembly</strong><br><strong>bwa mem -t 48 reference/${fasta} $myread/Ryegrass_SG700bp_trimmed_1P.fq $myread/Ryegrass_SG700bp_trimmed_2P.fq &gt; whatshap/Rabiosa_SG700.sam</strong></p><p><strong># convert sam to bam and sort it</strong><br><strong>samtools view -Sbh -@ 48 whatshap/Rabiosa_SG700.sam | samtools sort -@ 48 -o whatshap/Rabiosa_SG700_sorted.bam</strong></p><p><strong># call variants</strong><br><strong>bcftools mpileup -f reference/${fasta} whatshap/Rabiosa_SG700_sorted.bam | bcftools call -mv -Ov -o whatshap/Rabiosa_SG700.vcf</strong></p><p><strong># filter variants, quality &gt; 20 and only bi-allelic SNPs</strong><br><strong>bcftools filter -i 'QUAL&gt;20' whatshap/Rabiosa_SG700.vcf | bcftools view -g het -v snps -m2 -M2 &gt; whatshap/Rabiosa_SG700_f.vcf</strong></p><p><strong># extract vcf for all chr</strong><br><strong>bcftools view -t chr1,chr2,chr3,chr4,chr5,chr6,chr7 whatshap/Rabiosa_SG700_f.vcf &gt; whatshap/Rabiosa_SG700_f_chr.vcf</strong></p><p><strong># compress vcf and give new sample name</strong><br><strong>bgzip -c whatshap/Rabiosa_SG700_f_chr.vcf &gt; whatshap/Rabiosa_SG700_f_chr.vcf.gz</strong><br><strong>bcftools index whatshap/Rabiosa_SG700_f_chr.vcf.gz&nbsp;</strong></p><p><strong># reheader</strong><br><strong>bcftools reheader -s whatshap/<a href="http://sample.name/">sample.name</a> whatshap/Rabiosa_SG700_f_chr.vcf.gz &gt; whatshap/Rabiosa_SG700_f_chr_renamed.vcf.gz</strong></p><p><br></p><p>Of course, in addition to variant detection, we also need to map our ONT long reads to the reference. Instead of using minimap2, the most popular long-read mapping tool, we chose winnowmap, as it is said to map better, <a href="https://github.com/marbl/Winnowmap">https://github.com/marbl/Winnowmap</a></p><p><strong>#!/bin/bash</strong></p><p><strong># using winnowmap to align corrected ONT long reads to the phseduo-chromosomes</strong></p><p><strong>meryl count k=15 output merylDB ../reference/211209_Rabiosa_new_pseudomolecules_v1+unanchored_contigs.fasta</strong><br><strong>meryl print greater-than distinct=0.9998 merylDB &gt; repetitive_k15.txt</strong></p><p><strong>winnowmap -W repetitive_k15.txt -t 45 -R "@RG\tID:1\tSM:Rabiosa" -ax map-ont ../reference/211209_Rabiosa_new_pseudomolecules_v1+unanchored_contigs.fasta ../../Rabiosa_data/input/ONT_G4_q7a2kf.fq.gz --secondary=no &gt; output.sam</strong></p><p><strong>samtools sort -m 3G -@ 30 -o output_sorted.bam output.sam</strong></p><p><strong>samtools view -hb -F 2048 -@ 40 -o output_sorted_nosa.bam output_sorted.bam</strong></p><p><strong># extract chr</strong><br><strong>samtools view -hb -o output_sorted_nosa_chr.bam output_sorted_nosa.bam chr1 chr2 chr3 chr4 chr5 chr6 chr7</strong></p><p>Notice, when mapping long reads using winnowmap, we add read group name as Rabiosa, just to match the sample name in previous vcf files</p><p><br></p><p>Now, start whatshap:</p><p><strong>#!/bin/bash</strong></p><p><strong># read-based phasing with ONT long reads mapped by winnowmap</strong></p><p><strong>whatshap=/home/yutachen/.local/bin/whatshap</strong></p><p><strong># index alignment file</strong><br><strong>samtools index winnowmap/output_sorted_nosa_chr.bam</strong></p><p><strong># read-based phasing, long reads + hapcut2 global phase information</strong><br><strong>whatshap phase --sample Rabiosa -o winnowmap/phased.vcf --reference=reference/211209_Rabiosa_new_pseudomolecules_v1.fasta whatshap/Rabiosa_SG700_f_chr_renamed.vcf.gz \</strong><br><strong>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;winnowmap/output_sorted_nosa_chr.bam whatshap/Rabiosa_hapcut2_phased_chr_renamed.vcf.gz</strong></p><p><strong># get some statistics from phase blocks</strong><br><strong>whatshap stats --gtf winnowmap/phased.gtf --block-list winnowmap/phased.list --tsv winnowmap/phased_stats.tsv winnowmap/phased.vcf</strong></p><p><strong># need to index the phased vcf</strong><br><strong>bgzip -c winnowmap/phased.vcf &gt; winnowmap/phased.vcf.gz &amp;&amp; tabix winnowmap/phased.vcf.gz</strong></p><p><strong># tag reads based the phased SNPs</strong><br><strong>whatshap haplotag --reference=reference/211209_Rabiosa_new_pseudomolecules_v1.fasta --sample Rabiosa -o winnowmap/tagged.bam \</strong><br><strong>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; --output-haplotag-list winnowmap/tagged_reads.list winnowmap/phased.vcf.gz winnowmap/output_sorted_nosa_chr.bam</strong></p><p><strong># index the tagged bam</strong><br><strong>samtools index winnowmap/tagged.bam</strong></p><p><strong># extract reads based on read name</strong><br><strong># first get names from tagged_reads.list</strong></p><p><strong>grep "H1" winnowmap/tagged_reads.list | cut -f1 &gt; winnowmap/H1.list</strong><br><strong>grep "H2" winnowmap/tagged_reads.list | cut -f1 &gt; winnowmap/H2.list</strong><br><strong>grep "none" winnowmap/tagged_reads.list | cut -f1 &gt; winnowmap/untagged.list</strong></p><p><strong># extract all the read name</strong><br><strong>zcat ../Rabiosa_data/input/ONT_G4_q7a2kf.fq.gz | awk 'NR%4==1 {print}' | cut -f1 -d" " | sed 's/@//' &gt; winnowmap/all_read.list&nbsp;</strong></p><p><strong># find unmapped reads (unmapped to chrs)</strong><br><strong>cd winnowmap</strong><br><strong>./Find_unmapped_read.py</strong></p><p><strong># use seqtk to extract reads</strong><br><strong>ls winnowmap/H1.list winnowmap/H2.list winnowmap/untagged.list winnowmap/unmapped.list &gt; myreadlist</strong><br><strong>cat myreadlist | parallel -j 4 -k "seqtk subseq ../Rabiosa_data/input/ONT_G4_q7a2kf.fq.gz {} &gt; winnowmap/{/.}.fastq"</strong><br><strong>rm myreadlist</strong></p><p>What we did through above script is, first, build large and dense phase blocks with phase information provided by long reads and Hi-C, then tag long reads to different haplotypes based on SNPs they span, finally extract reads of different haplotypes based on read names</p><p><br></p><p>Phasing done! Tutorial of assembly with Flye can be seen from previous blog,&nbsp;<a class="confluence-link" href="https://unlimited.ethz.ch/pages/viewpage.action?pageId=63050497" data-linked-resource-id="63050497" data-linked-resource-version="3" data-linked-resource-type="blogpost" data-linked-resource-default-alias="Assemble ONT long reads using assembler Flye" data-base-url="https://unlimited.ethz.ch">Assemble ONT long reads using assembler Flye</a>. Tritex and ragtag deserve one independent blog for each, I will do that later.</p><p><br></p><p>Now, it's time to show some results</p><p>An example of read tagging from IGV, a region on Rabiosa chr6:</p><p><img class="confluence-embedded-image" draggable="false" src="./reference_based_phasing_files/Tag_reads.png" data-image-src="/download/attachments/128713540/Tag_reads.png?version=1&amp;modificationDate=1661348743647&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128714649" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="Tag_reads.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Tag_reads.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Tag_reads.png" data-image-height="924" data-image-width="1920"></p><p>In above plot, blue reads represent one haplotype, red reads represent the other haplotype, and gray reads are those cannot be tagged. It is very interesting to see that blue reads have a very large deletion (the very long black line) on the right side, which doesn't exist in the other haplotype, suggesting phasing or read tagging is correct. We can also zoom in to see more details for this region, for example below plot, we could see SNPs in blue reads have the same allele, and there are two small deletions in blue reads. Reads with the same alleles are tagged as the same haplotype, also suggesting phasing is correct!&nbsp;</p><p><img class="confluence-embedded-image" draggable="false" src="./reference_based_phasing_files/Tag_reads_zoom_in.png" data-image-src="/download/attachments/128713540/Tag_reads_zoom_in.png?version=1&amp;modificationDate=1661349387212&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128714673" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="Tag_reads_zoom_in.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Tag_reads_zoom_in.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Tag_reads_zoom_in.png" data-image-height="924" data-image-width="1920">&nbsp;</p><p>Now some statistics of phase blocks and tagged reads:</p><p><img class="confluence-embedded-image" draggable="false" src="./reference_based_phasing_files/phase_block.png" data-image-src="/download/attachments/128713540/phase_block.png?version=1&amp;modificationDate=1661499908877&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128716000" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="phase_block.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; phase_block.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; phase_block.png" data-image-height="400" data-image-width="800"></p><p>In above plot, the red line means the largest phase block on the chromosome and black lines are small phase block which are not integrated into the large block, and the blue vertical line indicates the total length of the chromosome. So, from this plot, we could see, by combining the global phase provided by Hi-C with local phase provided by long reads, every chromosome ended up with a chromosome-level phase block. However, not surprisingly, there are still small blocks&nbsp;&nbsp;</p><table class="wrapped confluenceTable"><tbody><tr><td class="confluenceTd">sample</td><td class="confluenceTd">chromosome</td><td class="confluenceTd">variants</td><td class="confluenceTd">phased</td><td class="confluenceTd">unphased</td><td class="confluenceTd">blocks</td><td class="confluenceTd">variant_per_block_max</td><td class="confluenceTd">bp_per_block_max</td><td class="confluenceTd">bp_per_block_sum</td><td class="confluenceTd">heterozygous_variants</td><td class="confluenceTd">heterozygous_snvs</td><td class="confluenceTd">phased_snvs</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr1</td><td class="confluenceTd">1064320</td><td class="confluenceTd">1056331</td><td class="confluenceTd">7989</td><td class="confluenceTd">776</td><td class="confluenceTd">1050411</td><td class="confluenceTd">262400071</td><td class="confluenceTd">264624494</td><td class="confluenceTd">1064320</td><td class="confluenceTd">1064320</td><td class="confluenceTd">1056331</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr2</td><td class="confluenceTd">1333495</td><td class="confluenceTd">1322687</td><td class="confluenceTd">10805</td><td class="confluenceTd">1042</td><td class="confluenceTd">1314780</td><td class="confluenceTd">341736942</td><td class="confluenceTd">345633772</td><td class="confluenceTd">1333495</td><td class="confluenceTd">1333495</td><td class="confluenceTd">1322687</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr3</td><td class="confluenceTd">1436084</td><td class="confluenceTd">1424888</td><td class="confluenceTd">11193</td><td class="confluenceTd">1010</td><td class="confluenceTd">1416897</td><td class="confluenceTd">367334000</td><td class="confluenceTd">371026499</td><td class="confluenceTd">1436084</td><td class="confluenceTd">1436084</td><td class="confluenceTd">1424888</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr4</td><td class="confluenceTd">1693703</td><td class="confluenceTd">1680821</td><td class="confluenceTd">12882</td><td class="confluenceTd">1380</td><td class="confluenceTd">1668435</td><td class="confluenceTd">432989167</td><td class="confluenceTd">442563544</td><td class="confluenceTd">1693703</td><td class="confluenceTd">1693703</td><td class="confluenceTd">1680821</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr5</td><td class="confluenceTd">1065333</td><td class="confluenceTd">1057183</td><td class="confluenceTd">8148</td><td class="confluenceTd">748</td><td class="confluenceTd">1052284</td><td class="confluenceTd">244874204</td><td class="confluenceTd">246243529</td><td class="confluenceTd">1065333</td><td class="confluenceTd">1065333</td><td class="confluenceTd">1057183</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr6</td><td class="confluenceTd">1092690</td><td class="confluenceTd">1084431</td><td class="confluenceTd">8258</td><td class="confluenceTd">783</td><td class="confluenceTd">1079145</td><td class="confluenceTd">257629914</td><td class="confluenceTd">259232995</td><td class="confluenceTd">1092690</td><td class="confluenceTd">1092690</td><td class="confluenceTd">1084431</td></tr><tr><td class="confluenceTd">Rabiosa</td><td class="confluenceTd">chr7</td><td class="confluenceTd">1359094</td><td class="confluenceTd">1347706</td><td class="confluenceTd">11388</td><td class="confluenceTd">919</td><td class="confluenceTd">1341820</td><td class="confluenceTd">339096701</td><td class="confluenceTd">341651603</td><td class="confluenceTd">1359094</td><td class="confluenceTd">1359094</td><td class="confluenceTd">1347706</td></tr></tbody></table><p><br></p><p>Check how many reads are tagged, untagged and unmapped to chromosomes. the *.list files are produced by the whatshap script above.</p><p><strong>seqkit stats H1.fastq H2.fastq untagged.fastq unmapped.fastq &gt; tag_reads.stats</strong></p><table class="wrapped confluenceTable"><tbody><tr><td class="confluenceTd">file</td><td class="confluenceTd">format</td><td class="confluenceTd">type</td><td class="confluenceTd">num_seqs</td><td class="confluenceTd">sum_len</td><td class="confluenceTd">min_len</td><td class="confluenceTd">avg_len</td><td class="confluenceTd">max_len</td></tr><tr><td class="confluenceTd">H1.fastq</td><td class="confluenceTd">FASTQ</td><td class="confluenceTd">DNA</td><td class="confluenceTd">3,703,084</td><td class="confluenceTd">49,511,773,743</td><td class="confluenceTd">2,000</td><td class="confluenceTd">13,370.4</td><td class="confluenceTd">381,245</td></tr><tr><td class="confluenceTd">H2.fastq</td><td class="confluenceTd">FASTQ</td><td class="confluenceTd">DNA</td><td class="confluenceTd">3,571,897</td><td class="confluenceTd">47,741,308,009</td><td class="confluenceTd">2,000</td><td class="confluenceTd">13,365.8</td><td class="confluenceTd">439,975</td></tr><tr><td class="confluenceTd">untagged.fastq</td><td class="confluenceTd">FASTQ</td><td class="confluenceTd">DNA</td><td class="confluenceTd">6,398,652</td><td class="confluenceTd">39,766,446,651</td><td class="confluenceTd">2,000</td><td class="confluenceTd">6,214.8</td><td class="confluenceTd">364,493</td></tr><tr><td class="confluenceTd">unmapped.fastq</td><td class="confluenceTd">FASTQ</td><td class="confluenceTd">DNA</td><td class="confluenceTd">2,003,066</td><td class="confluenceTd">16,474,467,909</td><td class="confluenceTd">2,000</td><td class="confluenceTd">8,224.6</td><td class="confluenceTd">335,711</td></tr></tbody></table><p><img class="confluence-embedded-image" draggable="false" src="./reference_based_phasing_files/total_length_reads_assigned.png" data-image-src="/download/attachments/128713540/total_length_reads_assigned.png?version=1&amp;modificationDate=1661521560756&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128716367" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="total_length_reads_assigned.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; total_length_reads_assigned.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; total_length_reads_assigned.png" data-image-height="600" data-image-width="400"></p><p>From above plot and table, we could see how much data were assigned to each read group. There are around 49 Gb reads are tagged as haplotype 1 group, which is around 25x haploid genome coverage and 47 Gb reads are assigned to haplotype 2 group, which is also around 25x haploid genome coverage. There are 39 Gb reads untagged and 16 Gb reads are not mapped to chromosomes.</p><p><br></p><p>With the haplotype 1 and haplotype 2 reads, I made two assemblies using Flye and raw assembly statistics are in below table. It looks like both assemblies are very fragmented and not complete, I think this is due to the relatively low coverage for each haplotype. So, based on this result, if we want to phase a genome using error-prone ONT data, then we need more data. I guess, we need 200x coverage data.</p><table class="wrapped confluenceTable"><tbody><tr><td class="confluenceTd"><p><strong>Assembly</strong></p></td><td class="confluenceTd"><p><strong>Total length</strong></p></td><td class="confluenceTd"><p><strong>Number of contigs</strong></p></td><td class="confluenceTd"><p><strong>N50</strong></p></td><td class="confluenceTd"><p><strong>N50n</strong></p></td></tr><tr><td class="confluenceTd"><p>FLYE_H1_miop_10k</p></td><td class="confluenceTd"><p>1.77 Gb</p></td><td class="confluenceTd"><p>11384</p></td><td class="confluenceTd"><p>275 kb</p></td><td class="confluenceTd"><p>1807</p></td></tr><tr><td class="confluenceTd"><p>FLYE_H2_miop_10k</p></td><td class="confluenceTd"><p>1.73 Gb</p></td><td class="confluenceTd"><p>11375</p></td><td class="confluenceTd"><p>265 kb</p></td><td class="confluenceTd"><p>1808</p></td></tr></tbody></table><p><br></p><p>Final assemblies for hap1 and hap2:</p><p><img class="confluence-embedded-image" draggable="false" height="400" src="./reference_based_phasing_files/Rabiosa_h1_kat.png" data-image-src="/download/attachments/128713540/Rabiosa_h1_kat.png?version=1&amp;modificationDate=1661524247959&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128716500" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="Rabiosa_h1_kat.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Rabiosa_h1_kat.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Rabiosa_h1_kat.png" data-image-height="1800" data-image-width="2400"><img class="confluence-embedded-image" draggable="false" height="400" src="./reference_based_phasing_files/Rabiosa_h2_kat.png" data-image-src="/download/attachments/128713540/Rabiosa_h2_kat.png?version=1&amp;modificationDate=1661524261847&amp;api=v2" data-unresolved-comment-count="0" data-linked-resource-id="128716503" data-linked-resource-version="1" data-linked-resource-type="attachment" data-linked-resource-default-alias="Rabiosa_h2_kat.png" data-base-url="https://unlimited.ethz.ch" data-linked-resource-content-type="image/png" data-linked-resource-container-id="128713540" data-linked-resource-container-version="9" title="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Rabiosa_h2_kat.png" data-location="MPB_BioInfo &gt; 2022/08/26 &gt; Read-based phasing or reference-based phasing!!! &gt; Rabiosa_h2_kat.png" data-image-height="1800" data-image-width="2400"></p><p>Hap1 BUSCO, C:89.7%[S:83.6%,D:6.1%],F:0.3%,M:10.0%,n:1614</p><p>Hap2 BUSCO, C:88.9%[S:83.6%,D:5.3%],F:0.6%,M:10.5%,n:1614</p>
        <p>&nbsp;</p>
    

<iframe id="d2VjMS7a" frameborder="0" src="./reference_based_phasing_files/translateSandbox.html" style="width: 0px; height: 0px; display: none;"></iframe></body></html>